src/engine/MultiColumnJoin.cpp

```diff
@@ -16,10 +17,12 @@ using std::string;
 // _____________________________________________________________________________
 MultiColumnJoin::MultiColumnJoin(QueryExecutionContext* qec,
                                  std::shared_ptr<QueryExecutionTree> t1,
-                                 std::shared_ptr<QueryExecutionTree> t2)
+                                 std::shared_ptr<QueryExecutionTree> t2,
+                                 bool allowSwappingChildrenOnlyForTesting)
     : Operation{qec} {
   // Make sure subtrees are ordered so that identical queries can be identified.
-  if (t1->getCacheKey() > t2->getCacheKey()) {
+  if (allowSwappingChildrenOnlyForTesting &&
+      t1->getCacheKey() > t2->getCacheKey()) {
```

Comment:
What is the reason for this? I don't think this was only for testing, but only to make the cache more efficient, did you have a probem with this?

Reply:
The name is the same one the join class uses. It is used to prevent swapping of children for the tests.